File: frontend/src/types/deal.ts
export enum DealType {
  clearance = "clearance",
  flash = "flash",
  happy_hour = "happy_hour",
}

export enum DealStatus {
  draft = "draft",
  active = "active",
  reserved = "reserved",
  expired = "expired",
  cancelled = "cancelled",
}

export interface Deal {
  id: string;
  store_id: string;
  product_id: string;
  deal_price: number;
  original_price: number;
  discount_pct: number;
  quantity_available: number;
  expiry_date: string;
  deal_type: DealType;
  status: DealStatus;
  listed_at: string | null;
  risk_score_at_listing: number;
  
  product_name?: string;
  store_name?: string;
  days_to_expiry?: number;
  is_urgent?: boolean;
}

export interface DealCreateRequest {
  product_id: string;
  deal_price: number;
  quantity_to_list: number;
  deal_type?: DealType;
}

export interface DealSuggestion {
  product_id: string;
  product_name: string;
  category: string;
  mrp: number;
  suggested_discount_pct: number;
  suggested_deal_price: number;
  risk_score: number;
  risk_label: "safe" | "watch" | "urgent" | "critical";
  days_to_expiry: number;
  quantity: number;
}

export interface DealListResponse {
  items: Deal[];
  total: number;
}

File: backend/app/schemas/deal.py
from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, ConfigDict, Field
import uuid

from app.core.constants import DealType, DealStatus

class DealCreateRequest(BaseModel):
    product_id: str
    deal_price: Decimal = Field(gt=0)
    quantity_to_list: int = Field(gt=0)
    deal_type: DealType = DealType.clearance
    
class DealOut(BaseModel):
    id: str | uuid.UUID
    store_id: str | uuid.UUID
    product_id: str | uuid.UUID
    deal_price: float
    original_price: float
    discount_pct: int
    quantity_available: int
    expiry_date: str
    deal_type: str | DealType
    status: str | DealStatus
    listed_at: str | datetime | None = None
    risk_score_at_listing: int
    
    product_name: str | None = None
    store_name: str | None = None
    days_to_expiry: int | None = None
    is_urgent: bool | None = None
    
    model_config = ConfigDict(from_attributes=True)
    
class DealSuggestion(BaseModel):
    product_id: str | uuid.UUID
    product_name: str
    category: str
    mrp: float
    suggested_discount_pct: int
    suggested_deal_price: float
    risk_score: int
    risk_label: str
    days_to_expiry: int
    quantity: int
    
class DealListResponse(BaseModel):
    items: list[DealOut]
    total: int
